docs: feature-request: add aliases for short codes and add new format mode

It would be great if for every rule that has a page in the documentation you could go to it not only by its full name (`unused-import`: https://beta.ruff.rs/docs/rules/unused-import/ ) but also by its short code (`F401`: https://beta.ruff.rs/docs/rules/F401/). 
This would make it easier to find best practice examples for this bug.

Also, I suggest adding a separate formatting mode based on `text` (e.g., call it `text_links`). In which a link to such pages would be given next to the short error code. For example, it could look like this:
```
web/currency/models.py:1:1: I001 (https://beta.ruff.rs/docs/rules/I001/) [*] Import block is un-sorted or un-formatted
web/currency/models.py:10:7: N818 (https://beta.ruff.rs/docs/rules/N818/) Exception name `CurrencyNotFound` should be named with an Error suffix
web/currency/models.py:20:121: E501 (https://beta.ruff.rs/docs/rules/E501/) Line too long (152 > 120 characters)
```

Or, for example, add a separate setting to enable link display mode, which will add links for formatting modes: `text`, `grouped`

This would make it a lot easier to find "how to do it better" examples and links to the original rules from the linters (which are on the rules pages) and would speed up the process of correcting mistakes.

PS: In option `format` there is no `groupped` in the default values.

Long names are great too, but they will take up a lot of space in the links (if implemented) within the error.

I saw this idea in the linter [ShellCheck for Visual Studio Code](https://marketplace.visualstudio.com/items?itemName=timonwong.shellcheck), when hovering over the error in the IDE - a link is displayed (in the example it is SC2086 on the right) and when you click on it we get to the page: https://www.shellcheck.net/wiki/SC2086
<img width="659" alt="Снимок экрана 2023-05-30 в 00 32 56" src="https://github.com/charliermarsh/ruff/assets/945306/a7207a82-c3c4-4cd5-a261-186da7caf591">


This is very handy: if you don't know what is the right fix, go to the documentation for the rule and see examples.

PS: or make a subdomain with short links that will redirect to long links. For example: `https://r.ruff.rs/F401` will redirect to `https://beta.ruff.rs/docs/rules/unused-import/`

--- a/docs/requirements-insiders.txt
+++ b/docs/requirements-insiders.txt
@@ -2,3 +2,4 @@ PyYAML==6.0.1
 black==23.10.0
 mkdocs==1.5.0
 git+ssh://git@github.com/astral-sh/mkdocs-material-insiders.git@38c0b8187325c3bab386b666daf3518ac036f2f4
+mkdocs-redirects==1.2.1

--- a/docs/requirements.txt
+++ b/docs/requirements.txt
@@ -2,3 +2,4 @@ PyYAML==6.0.1
 black==23.10.0
 mkdocs==1.5.0
 mkdocs-material==9.1.18
+mkdocs-redirects==1.2.1

--- a/mkdocs.insiders.yml
+++ b/mkdocs.insiders.yml
@@ -1,4 +1,1 @@
 INHERIT: mkdocs.generated.yml
-plugins:
-  - search
-  - typeset

--- a/mkdocs.template.yml
+++ b/mkdocs.template.yml
@@ -60,6 +60,7 @@ markdown_extensions:
       alternate_style: true
 plugins:
   - search
+  - typeset
 extra_css:
   - stylesheets/extra.css
 not_in_nav: |

--- a/scripts/generate_mkdocs.py
+++ b/scripts/generate_mkdocs.py
@@ -2,6 +2,7 @@
 from __future__ import annotations
 
 import argparse
+import json
 import re
 import shutil
 import subprocess
@@ -138,9 +139,37 @@ def main() -> None:
 
             f.write(clean_file_content(file_content, title))
 
-    # Add the nav section to mkdocs.yml.
     with Path("mkdocs.template.yml").open(encoding="utf8") as fp:
         config = yaml.safe_load(fp)
+
+    # Add the redirect section to mkdocs.yml.
+    rules = json.loads(
+        subprocess.check_output(
+            [
+                "cargo",
+                "run",
+                "-p",
+                "ruff_cli",
+                "--",
+                "rule",
+                "--all",
+                "--output-format",
+                "json",
+            ],
+        ),
+    )
+    config["plugins"].append(
+        {
+            "redirects": {
+                "redirect_maps": {
+                    f'rules/{rule["code"]}.md': f'rules/{rule["name"]}.md'
+                    for rule in rules
+                },
+            },
+        },
+    )
+
+    # Add the nav section to mkdocs.yml.
     config["nav"] = [{section.title: section.filename} for section in SECTIONS]
 
     with Path("mkdocs.generated.yml").open("w+") as fp:

